### DISCUSSION TITLE
The DetailedList() method in Toga version 0.3.1 is working properly. After upgrading to version 0.4.2, there was an issue when calling the on_select parameter on the Android side, but everything is working fine on the Windows side.

### DISCUSSION THREAD

**Discussion by jlij** [beeware/toga#2461] [Q&A] [answered]
--------- beginning of crash
E/AndroidRuntime: FATAL EXCEPTION: main
E/AndroidRuntime: Process: vip.eicp.patrol, PID: 31258
E/AndroidRuntime: com.chaquo.python.PyException: TypeError: patrol.on_select_car() got an unexpected keyword argument 'row'
E/AndroidRuntime:       at <python>.toga.handlers._handler(handlers.py:66)
E/AndroidRuntime:       at <python>.toga_android.widgets.detailedlist.onClick(detailedlist.py:34)
E/AndroidRuntime:       at <python>.chaquopy_java.call(chaquopy_java.pyx:354)
E/AndroidRuntime:       at <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrowsNative(chaquopy_java.pyx:326)
E/AndroidRuntime:       at com.chaquo.python.PyObject.callAttrThrowsNative(Native Method)
E/AndroidRuntime:       at com.chaquo.python.PyObject.callAttrThrows(PyObject.java:232)
E/AndroidRuntime:       at com.chaquo.python.PyInvocationHandler.invoke(PyInvocationHandler.java:31)
E/AndroidRuntime:       at java.lang.reflect.Proxy.invoke(Proxy.java:913)
E/AndroidRuntime:       at $Proxy8.onClick(Unknown Source)
E/AndroidRuntime:       at android.view.View.performClick(View.java:6291)
E/AndroidRuntime:       at android.view.View$PerformClick.run(View.java:24931)
E/AndroidRuntime:       at android.os.Handler.handleCallback(Handler.java:808)
E/AndroidRuntime:       at android.os.Handler.dispatchMessage(Handler.java:101)
E/AndroidRuntime:       at android.os.Looper.loop(Looper.java:166)
E/AndroidRuntime:       at android.app.ActivityThread.main(ActivityThread.java:7529)
E/AndroidRuntime:       at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime:       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
E/AndroidRuntime:       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
I/Process : Sending signal. PID: 31258 SIG: 9

**Comment by freakboy3742** [comment]
When reporting an error, a critical part of the report is telling us *what you did to generate the error*. Without sample code, it's almost impossible to resolve any issue.

In this case, the error suggests that you haven't adapted to one of the backwards incompatible changes in the 0.4 release - specifically, the change in prototype for the `on_select` handler no longer accepts a `row` argument. This is specifically called out in the [Backwards incompatibility notes for the 0.4.0 release](https://toga.readthedocs.io/en/stable/background/project/releases.html#id11).

**Reply by jlij** [reply]
Code from https://github.com/beeware/toga/tree/main/examples/detailedlist/detailedlist
No changes have been made

**Reply by freakboy3742** [reply]
You are *not* running that code. The handler in your stack trace is `patrol.on_select_car()`. That is not the name of the [on_select handler in that example app](https://github.com/beeware/toga/blob/main/examples/detailedlist/detailedlist/app.py#L12).

**Reply by jlij** [reply]
--------- beginning of crash
E/AndroidRuntime: FATAL EXCEPTION: main
E/AndroidRuntime: Process: vip.eicp.patrol, PID: 17685
E/AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{vip.eicp.patrol/org.beeware.android.MainActivity}: com.chaquo.python.PyException: TypeError: DetailedList.__init__() got an unexpected keyword argument 'missing_value'
E/AndroidRuntime:       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3303)
E/AndroidRuntime:       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3411)
E/AndroidRuntime:       at android.app.ActivityThread.-wrap12(Unknown Source:0)
E/AndroidRuntime:       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1994)
E/AndroidRuntime:       at android.os.Handler.dispatchMessage(Handler.java:108)
E/AndroidRuntime:       at android.os.Looper.loop(Looper.java:166)
E/AndroidRuntime:       at android.app.ActivityThread.main(ActivityThread.java:7529)
E/AndroidRuntime:       at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime:       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
E/AndroidRuntime:       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
E/AndroidRuntime: Caused by: com.chaquo.python.PyException: TypeError: DetailedList.__init__() got an unexpected keyword argument 'missing_value'
E/AndroidRuntime:       at <python>.patrol.app.startup(app.py:105)
E/AndroidRuntime:       at <python>.toga_android.app.create(app.py:175)
E/AndroidRuntime:       at <python>.toga_android.app.main_loop(app.py:186)
E/AndroidRuntime:       at <python>.toga.app.main_loop(app.py:517)
E/AndroidRuntime:       at <python>.__main__.<module>(__main__.py:4)
E/AndroidRuntime:       at <python>.runpy._run_code(<frozen runpy>:88)
E/AndroidRuntime:       at <python>.runpy._run_module_code(<frozen runpy>:98)
E/AndroidRuntime:       at <python>.runpy.run_module(<frozen runpy>:226)
E/AndroidRuntime:       at <python>.chaquopy_java.call(chaquopy_java.pyx:354)
E/AndroidRuntime:       at <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrowsNative(chaquopy_java.pyx:326)
E/AndroidRuntime:       at com.chaquo.python.PyObject.callAttrThrowsNative(Native Method)
E/AndroidRuntime:       at com.chaquo.python.PyObject.callAttrThrows(PyObject.java:232)
E/AndroidRuntime:       at com.chaquo.python.PyObject.callAttr(PyObject.java:221)
E/AndroidRuntime:       at org.beeware.android.MainActivity.onCreate(MainActivity.java:85)
E/AndroidRuntime:       at android.app.Activity.performCreate(Activity.java:7383)
E/AndroidRuntime:       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1218)
E/AndroidRuntime:       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3256)
E/AndroidRuntime:       ... 9 more
I/Process : Sending signal. PID: 17685 SIG: 9

**Reply by freakboy3742** [reply]
That is a different error, and *still* doesn't correspond to the example app you've referenced. 

And, again, the error message is telling you what is wrong - ``DetailedList.__init__() got an unexpected keyword argument 'missing_value'``. This is a valid argument in 0.4.2, so if you're getting this message, it means you're not using Toga 0.4.2 when the code is running on Android. This is likely because your dependencies haven't been updated.

**Reply by jlij** [reply, marked as answer]
Sorry, after carefully reading the example you provided, we found that an important parameter * * kwargs is missing. After adding it, it worked fine. Thank you very much for your reply.
